mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp

```diff
@@ -173,14 +173,62 @@ class SelectOpConversion : public OpConversionPattern<arith::SelectOp> {
   }
 };
 
+// Floating-point to integer, integer to floating-point conversions.
+template <typename CastOp, bool opHasTruncateSemantics>
+class CastOpConversion : public OpConversionPattern<CastOp> {
+private:
+  bool floatToIntTruncates;
+
+public:
+  CastOpConversion(const TypeConverter &typeConverter, MLIRContext *context,
+                   bool optionFloatToIntTruncates)
+      : OpConversionPattern<CastOp>(typeConverter, context),
+        floatToIntTruncates(optionFloatToIntTruncates) {}
+
+  LogicalResult
+  matchAndRewrite(CastOp castOp, typename CastOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Type operandType = adaptor.getIn().getType();
+    if (!operandType.isIntOrFloat())
+      return rewriter.notifyMatchFailure(castOp,
+                                         "casts are only supported on scalars");
```

Comment:
This will be covered by the `isSupportedFloat/IntegerType`, so you can drop this.

Reply:
Good remark!